### PR TITLE
Convert some more vars to knobs and remove legacy code

### PR DESCRIPTION
**Pull request by damccorm**
agent.AllowAllEndpoints/SecureFiles were never getting set so its safe to remove them (there was a todo to remove them a long time ago)

### CODE REVIEW

File: src/Microsoft.VisualStudio.Services.Agent/HostContext.cs [hunk 1]
```diff
@@ -134,7 +134,7 @@ public HostContext(string hostType, string logFile = null)
 
             // Enable Http trace
             bool enableHttpTrace;
-            if (bool.TryParse(Environment.GetEnvironmentVariable("VSTS_AGENT_HTTPTRACE"), out enableHttpTrace) && enableHttpTrace)
+            if (AgentKnobs.HttpTrace.GetValue(this).AsBoolean());
```
Comment (jtpetty) [inline src/Microsoft.VisualStudio.Services.Agent/HostContext.cs:137]: I am guessing you did not mean to have that trailing semi-colon?

File: src/Microsoft.VisualStudio.Services.Agent/TraceSetting.cs [hunk 2]
```diff
@@ -5,19 +5,23 @@
...
 using System.Runtime.Serialization;
+using Agent.Sdk.Knob;
+using Microsoft.VisualStudio.Services.Agent.Util;
 
 namespace Microsoft.VisualStudio.Services.Agent
 {
     [DataContract]
     public class TraceSetting
     {
+        private static UtilKnobValueContext _knobContext = new UtilKnobValueContext();
```
Comment (jtpetty) [inline src/Microsoft.VisualStudio.Services.Agent/TraceSetting.cs:16]: Now that I am seeing this more, should UtilKnobValueContext be a singleton?

Comment (damccorm) [reply src/Microsoft.VisualStudio.Services.Agent/TraceSetting.cs]: Good idea! Done

File: src/Microsoft.VisualStudio.Services.Agent/HostContext.cs (hunk 1 above)
Comment (damccorm) [reply src/Microsoft.VisualStudio.Services.Agent/HostContext.cs]: Correct

**Pull request merged**